lib/hexpm/web/views/api/package_view.ex

```diff
@@ -25,5 +25,10 @@ defmodule Hexpm.Web.API.PackageView do
     |> include_if_loaded(:releases, package.releases, ReleaseView, "minimal.json", package: package)
     |> include_if_loaded(:downloads, package.downloads, DownloadView, "show.json")
     |> include_if_loaded(:owners, package.owners, UserView, "minimal.json")
+    |> group_downloads()
+  end
+
+  defp group_downloads(%{downloads: downloads} = package) do
+    Map.put(package, :downloads, Enum.reduce(downloads, %{}, &Map.merge(&1, &2)))
```

Comment:
All the views are written under the assumption that no associations are guaranteed. It makes it easier to use the views from different controllers that doesn't load the same data.